airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -73,191 +73,129 @@
+      prepareForNextRunAndContinueAsNew(connectionUpdaterInput);
     }
   }
 
-  private void reportSuccess(final ConnectionUpdaterInput connectionUpdaterInput) {
+  private CancellationScope generateSyncWorkflowRunnable(ConnectionUpdaterInput connectionUpdaterInput) {
+    return Workflow.newCancellationScope(() -> {
+      connectionId = connectionUpdaterInput.getConnectionId();
+
+      if (connectionUpdaterInput.getWorkflowState() != null) {
+        workflowState = connectionUpdaterInput.getWorkflowState();
+      }
+
+      workflowState.setResetConnection(connectionUpdaterInput.isResetConnection());
+
+      Duration timeToWait = getTimeToWait(connectionUpdaterInput.getConnectionId());
+
+      Workflow.await(timeToWait,
+          () -> skipScheduling() || connectionUpdaterInput.isFromFailure());
+
+      val needToProcessSync = !workflowState.isUpdated() && !workflowState.isDeleted();
+
+      if (needToProcessSync) {
+        jobId = getOrCreateJobId(connectionUpdaterInput);
+
+        attemptId = createAttemptId(jobId);
+
+        final GeneratedJobInput jobInputs = getJobInput();
+
+        reportJobStarting();
+        StandardSyncOutput standardSyncOutput = null;
+        try {
+          standardSyncOutput = runChildWorkflow(jobInputs);
+
+          workflowState.setFailed(getFailStatus(standardSyncOutput));
+
+          if (workflowState.isFailed()) {
+            reportFailure(connectionUpdaterInput, standardSyncOutput);
+            prepareForNextRunAndContinueAsNew(connectionUpdaterInput);
+          } else {
+            reportSuccess(connectionUpdaterInput, standardSyncOutput);
+            prepareForNextRunAndContinueAsNew(connectionUpdaterInput);
+          }
+        } catch (final ChildWorkflowFailure childWorkflowFailure) {
+          if (childWorkflowFailure.getCause() instanceof CanceledFailure) {
+            // do nothing, cancellation handled by cancellationScope
+
+          } else if (childWorkflowFailure.getCause() instanceof ActivityFailure) {
+            final ActivityFailure af = (ActivityFailure) childWorkflowFailure.getCause();
+            failures.add(FailureHelper.failureReasonFromWorkflowAndActivity(
+                childWorkflowFailure.getWorkflowType(),
+                af.getActivityType(),
+                af.getCause(),
+                jobId,
+                attemptId));
+            reportFailure(connectionUpdaterInput, standardSyncOutput);
+          } else {
+            failures.add(
+                FailureHelper.unknownOriginFailure(childWorkflowFailure.getCause(), jobId, attemptId));
+            reportFailure(connectionUpdaterInput, standardSyncOutput);
+          }
+        }
+      }
+    });
+  }
+
+  private void reportSuccess(final ConnectionUpdaterInput connectionUpdaterInput, StandardSyncOutput standardSyncOutput) {
     workflowState.setSuccess(true);
     runMandatoryActivity((input) -> jobCreationAndStatusUpdateActivity.jobSuccess(input), new JobSuccessInput(
```

Comment:
```suggestion
    runMandatoryActivity(jobCreationAndStatusUpdateActivity::jobSuccess, new JobSuccessInput(
```

Reply:
Done, Sorry I missed it was changing the other ones